mp4/box.go

```diff
@@ -173,13 +170,10 @@ func DecodeBox(startPos uint64, r io.Reader) (Box, error) {
 		log.Debugf("Found supported box %v, size %v", h.name, h.size)
 		b, err = d(h, startPos, io.LimitReader(r, remainingLength))
 	}
-	if h.size != b.Size() {
-		log.Errorf("### Mismatch size %d %d for %s", h.size, b.Size(), b.Type())
-	}
 	if err != nil {
-		log.Errorf("Error while decoding %s : %s", h.name, err)
-		return nil, err
+		return nil, errors.Wrap(err, fmt.Sprintf("couldn't decode %s", h.name))
```

Comment:
Why not use the wrapping functionality introduce in 1.13 instead of adding a third-party library? See https://blog.golang.org/go1.13-errors for more information.

Reply:
Sounds like a good idea. Will do that. Thanks.